pkg/cmd/create/create_devpod.go

```diff
@@ -238,19 +252,27 @@ func (o *CreateDevPodOptions) Run() error {
 	podResources := client.CoreV1().Pods(ns)
 	var exposeServicePorts []int
 
-	gitLabelKey := ""
-	gitLabelValue := ""
+	gitSchemeLabelKey := ""
```

Comment:
That's a lot of variables. What do you think about introducing a struct which contains all the relevant labels? Something like _devPodLabels_. Then we can encapsulate all required variables in there and we can have something like `func (d *_devPodLabels_) populateFromGitInfo( *gits.GitRepository) error` to set the fields. 
We could also have `func (d *_devPodLabels_) matchLabels() map[string]string`. And maybe even `func (d *_devPodLabels_) addLabels(pod *corev1.Pod) `. WDYT?

Reply:
It's ok for me .. working on it !

Reply:
First step: populateFromGitInfo and getLabels it's ok ... but addLabels I'm not sure .. You must check if you modify the pod labels before update, sou you need to compare the pod labels with the merged labels (pod lobales + git labels) .. maybe `func (d *_devPodLabels_) mergeLabels( labels map[string]string) map[string]string` is interesting, but not addLabels(pod)

Reply:
Up to you, these were just suggestions initially. Hard to tell the details w/o actually cutting the code.